### PR TITLE
feat: add SMS vendor client API

### PR DESCRIPTION
**Pull request by johanseto**
## Description
feat: add SMS vendor client API

## Testing instructions
Set the following settings.
``` json
"SMS_VENDOR_URL": "https://example.com",
"SMS_VENDOR_SEND_SMS_PATH": "sendSMS", (Optional)
"SMS_VENDOR_USER": "test-user",
"SMS_VENDOR_PASSWORD" : "test-password",
"SMS_VENDOR_MSG_SENDER": "test-sender" (Optional)
```
The testing is waiting for the configuration of the vendor URL.

## Additional information

**Jira story**
https://edunext.atlassian.net/browse/FUTUREX-692

## Checklist for Merge

- [ ] Tested in a remote environment
- [ ] Updated documentation
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

File: eox_nelp/api_clients/__init__.py [hunk 1]
```diff
@@ -186,3 +186,20 @@ def _authenticate(self):
         session.auth = HTTPBasicAuth(self.user, self.password)
 
         return session
+
+
+class AbstractBasicApiClient(AbstractApiClient):
```
Comment (andrey-canon) [inline eox_nelp/api_clients/__init__.py:191]:
```suggestion
class AbstractNonAuthenticatedApiClient(AbstractApiClient):
```

File: eox_nelp/api_clients/__init__.py [hunk 2]
```diff
@@ -186,3 +186,20 @@ def _authenticate(self):
...
+    """Abstract class for a basic  API client without authentication.
+    This class provides basic functionality for an API client that requires
+    basic api client without auth. Use a normal session.
+    """
+    def _authenticate(self):
+        """Authenticate the session with the user and password.
+        Creates and configures a requests session without authentication.
+        Returns:
```
Comment (andrey-canon) [inline eox_nelp/api_clients/__init__.py:199]:
```suggestion
        """Creates and configures a requests session without authentication.

        Returns:
```

File: eox_nelp/api_clients/__init__.py [hunk 3]
```diff
@@ -186,3 +186,20 @@ def _authenticate(self):
...
+            requests.Session: Basic Session.
+        """
+        # pylint: disable=no-member
+        session = requests.Session()
+
+        return session
```
Comment (andrey-canon) [inline eox_nelp/api_clients/__init__.py:205]:
```suggestion
        return requests.Session()
```

File: eox_nelp/api_clients/sms_vendor.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        @audit_method(action="Send SMS with sms_vendor API")
+        def send_sms_request(recipient, message):
+            """This is a wrapper that allows to make audit-able the send_SMS method."""
+            path = getattr(settings, "SMS_VENDOR_SEND_SMS_PATH", "")
+            payload = {
+                "message": message,
+                "number": recipient,
+                "username": getattr(settings, "SMS_VENDOR_USERNAME"),
+                "password": getattr(settings, "SMS_VENDOR_PASSWORD"),
+                "sender": "NELC",
```
Comment (andrey-canon) [inline eox_nelp/api_clients/sms_vendor.py:44]: could you make this field configurable ?

**Review by andrey-canon — CHANGES_REQUESTED** [review]

File: eox_nelp/api_clients/sms_vendor.py (hunk 4 above)
Comment (johanseto) [reply eox_nelp/api_clients/sms_vendor.py]: 7e82bf4afcba87f8cfc4a07527ff57e78a4d2479

File: eox_nelp/api_clients/__init__.py (hunk 1 above)
Comment (johanseto) [reply eox_nelp/api_clients/__init__.py]: da3d81bbad02880aca74ed07e028f66cb09d6e1f

**Review by andrey-canon — APPROVED** [review]

**Pull request merged**